### ISSUE TITLE
Issues with HvDatePicker Component

### ISSUE THREAD

**Issue by Quernest** [lumada-design/hv-uikit-react#3935]
### Which version of NEXT UI Kit are you opening this bug for?

v5.x

### Latest version

- [X] I have tested the latest version

### No Hitachi confidential content

- [X] I'm not including any Hitachi confidential content

### Current behavior 😯

**Bug Description:**

**Placeholder Overflow:** When using the `HvDatePicker` component, I observed that if the placeholder text is long, it does not get truncated and extends beneath the calendar icon.
The same issue with placeholder for `HvDropdown` by the way.

**Missing Clear Icon:** The `showClear` prop does not add the clear icon to the input

**Prop onChange is called on close**: when I am closing date picker onChange is called. Why?

**Inability to Clear Input:** Additionally, there is an issue where the input cannot be cleared again by selecting the same date on the calendar (maybe not a bug but `HvDropdown` works this way)

Example:
<img width="182" alt="Screenshot 2024-01-05 at 15 42 44" src="https://github.com/lumada-design/hv-uikit-react/assets/21086792/9f40b6e0-508d-4596-af34-a91d38dfa15b">

### Expected behavior 🤔

1. The long placeholder text should be truncated and not extend beneath the calendar icon.
2. The `showClear` prop should add the clear icon to the input (calendar's input has it)
3. The input should be cleared when selecting the same date on the calendar (?)
4. `onChange` should not be called when I am closing date picker

### Steps to reproduce 🕹

1. Use the HvDatePicker component with a long placeholder text.
2. Set the showClear prop to true.
3. Select a date on the calendar.
5. Attempt to clear the input by selecting the same date again (or using clear icon)
6. Listen `onChange` and close date picker

```tsx
<HvDatePicker
  locale="en-US"
  status="standBy"
  disablePortal={false}
  value={value}
  onChange={handleChange}
  placeholder={`Filter looooooooooooooooooooooong column header name`}
  showClear
/>
```

**Comment by MEsteves22** [comment]
Hello @Quernest 👋

We'll take a look at the issues you raised and let you know when fixes are available. Thanks!

**Comment by MEsteves22** [comment]
Regarding the `showClear` prop, the clear button will be visible when the date picker is in range mode (`rangeMode`) or when actions are enabled (`showActions`). You can find an example [here](https://lumada-design.github.io/uikit/master/?path=/docs/components-date-picker--docs) on the "With Selection List" sample.
Is this enough for your use case? 

Regarding the inability to clear the input by selecting the same date again, this is intentional at the moment.

**Comment by MEsteves22** [comment]
Hello @Quernest 👋

The issues related with `onChange` and the `placeholder` should be fixed on @hitachivantara/uikit-react-core@5.44.1. Thanks!

**Comment by MEsteves22** [comment]
Hello @Quernest 👋

We'll be closing this issue for now since the bugs were fixed. Let us know if you need anything else. Thanks!
